scripts/vagrant-base-setup.sh

```diff
@@ -2,13 +2,15 @@
 
 set -xe
 
-# python3-virtualenv is not available on Trusty so install it here
-sudo -E apt-get -y install python3-setuptools python3-dev python3-virtualenv
+sudo -E apt-get -y install python3-setuptools python3-dev python3-virtualenv python3-pip
 
 # create a virtualenv and install lxdock
 python3 -m virtualenv -p python3 ~/venv
+pip3 install pip
```

Comment:
Should this be "pip3 install -U pip"? so that it upgrades to the latest

Reply:
To be honest I don't quite understand why the statement "pip3 install pip" is there as we are creating a virtualenv the line above it (on line 8) and then activating it on line 10. What I would have thought to see is upgrading pip just after activating the virtualenv maybe? But why install pip with pip3 globally I wonder what the purpose is for that?

Reply:
The problem was, that the binary name in Ubuntu for the python3 version of pip is `pip3`, but the Makefile calls just `pip`. So it was either changing the Makefile, creating a symlink to pip3 or installing it in the virtualenv.

Reply:
Ok that is really odd because as far as I know as soon as you activate the virtualenv, pip is available from venv/bin/pip regardless if python3-pip is installed from apt. I can have a play with it and see but from my perspective it seems odd to do any pip install commands before the virtualenv is activated, they should appear after the virtualenv is activated so that we know we are using pip from the virtualenv not the system /usr/bin or /usr/local/bin

Reply:
I'm pretty sure the way it's activating the virtualenv now doesn't work like that in scripts that is probably why if you use "pip" it won't find it because the path isn't set but I need to verify this. From my experience, using source like that in a script doesn't work

Reply:
You are right. I just tested it on a fresh machine, it is really not necessary. I will remove it.

Reply:
1a3fa4f78396031c9947e373d132a87dc4ba11af